### PR TITLE
feat(lambda): layer version removal policy

### LINKED ISSUE #12718: (aws-lambda) Layers older versions disappear when deploy a new version
When I try to update a new lambda layer version, the old version disappear

### Reproduction Steps

```
    const layer = new lambda.LayerVersion( stack, "my_layer", {
        code: lambda.Code.fromAsset(path.join(__dirname, "./path/to/lib/nodejs.zip")),
        compatibleRuntimes: [lambda.Runtime.NODEJS_12_X]
    });
    const lambda = new lambda.Function( stack, "my_lambda", { 
        runtime: lambda.Runtime.NODEJS_12_X,
        code: lambda.Code.fromAsset(`lambda_dir`),
        handler: "index.handler",
        environment: {
            some: env,
        },
        layers: [layer]        
    });
```

### What did you expect to happen?

I have versions 1 - n available to me in this layer

### What actually happened?

Only the recent deployed version available
Screenshot 2021-01-27 at 11 22 25 AM

### Environment

  - **CDK CLI Version  :*1.86.0*
  - **Framework Version:**
  - **Node.js Version:** 12.19.0 -->
  - **OS               :*mac 10.15.7*
  - **Language (Version):**TypeScript (4.0.5) -->

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by swar8080**
closes #12718

Allows providing a removal policy for the  `Lambda::LayerVersion` L2 construct to retain previous layer versions when the layer is updated. When a value is not provided, the default CloudFormation behavior is used, which is to delete the previous version defined in the stack.

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/lib/layers.ts [hunk 1]
```diff
@@ -28,6 +28,14 @@ export interface LayerVersionOptions {
    * @default - A name will be generated.
    */
   readonly layerVersionName?: string;
+
+  /**
+   * Whether to retain the previous version of the layer when a new version is added
+   * or when the stack is deleted.
+   *
+   * @default - The previous version is deleted.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/lib/layers.ts:36]: I believe it should be this -

```suggestion
   * @default RemovalPolicy.DESTROY
```

File: packages/@aws-cdk/aws-lambda/lib/layers.ts [hunk 2]
```diff
@@ -28,6 +28,14 @@ export interface LayerVersionOptions {
    * @default - A name will be generated.
    */
   readonly layerVersionName?: string;
+
+  /**
+   * Whether to retain the previous version of the layer when a new version is added
+   * or when the stack is deleted.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/lib/layers.ts:34]:
```suggestion
   * Whether to retain this version of the layer when a new version is added
   * or when the stack is deleted.
```

File: packages/@aws-cdk/aws-lambda/README.md [hunk 3]
```diff
@@ -171,7 +171,7 @@ fn.currentVersion.addAlias('live');
 
 ## Layers
 
-The `lambda.LayerVersion` class can be used to define Lambda layers and manage
+The `lambda.LayerVersion` class can be used to define Lambda layer versions and manage
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/README.md:174]: Revert this change and add a statement at the end of this section about configuring removal policies.

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for submitting this PR. The code looks good.

I have a few comments around the documentation.

File: packages/@aws-cdk/aws-lambda/README.md (hunk 3 above)
Comment (swar8080) [reply packages/@aws-cdk/aws-lambda/README.md]: I added better documentation to the README. Let me know if it's too much detail

File: packages/@aws-cdk/aws-lambda/README.md [hunk 4]
```diff
@@ -176,6 +176,17 @@ granting permissions to other AWS accounts or organizations.
 
 [Example of Lambda Layer usage](test/integ.layer-version.lit.ts)
 
+### Layer Versioning and Removal Policies
+
+By default, updating a layer creates a new version and deletes the current version managed by the stack. Alternatively, a removal policy can be used to retain the current version:
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/README.md:181]: Something like this and I don't think you need a separate section for this - 

```suggestion
By default, updating a layer creates a new layer version, and CloudFormation will delete the old version as part of the stack update.
Alternatively, a removal policy can be used to retain the current version:
```

**Review by nija-at — DISMISSED** [review]

File: packages/@aws-cdk/aws-lambda/README.md (hunk 4 above)
Comment (swar8080) [reply packages/@aws-cdk/aws-lambda/README.md]: that sounds much clearer, thanks

**Review by nija-at — APPROVED** [review]
